crates/amaru-ledger/src/context.rs

```diff
@@ -211,7 +211,7 @@ pub trait DRepsSlice {
         anchor: Option<Anchor>,
     ) -> Result<(), UpdateError<StakeCredential>>;
 
-    fn unregister(&mut self, drep: StakeCredential, refund: Lovelace);
+    fn unregister(&mut self, drep: StakeCredential, refund: Lovelace, pointer: CertificatePointer);
```

Comment:
This is required so that we can track the precise moment drep deregisters, in order to re-implement the v9 bug where one could delegate to a non-existing DRep. 

Without tracking the de-registration time (i.e. if we only look at re-registration times), we cannot properly prune delegation from before the de-registration (which ends when de-registering) from those that happened after the re-registration (which will be immediately effective on the re-registration).